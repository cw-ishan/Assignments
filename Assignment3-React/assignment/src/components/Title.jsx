import React from 'react'
import { useSelector } from 'react-redux'
const styles = {
  fontSize:'30px'
}

function Title() {
    let value=0;
    value=useSelector(state=>state.products.length)
    return (
    <h1 style={styles} data-testid="title1">
       {value} Used Cars in India
    </h1>
  )
}

export default Title