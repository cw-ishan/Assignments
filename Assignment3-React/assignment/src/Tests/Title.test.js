import { render, screen } from '@testing-library/react';
import { useSelector } from 'react-redux'
import Title from '../components/Title';
jest.mock('react-redux')

beforeEach(()=>{
    useSelector.mockReturnValue("32");
})

test('snapshot testing with mocking useSelector',()=>{
    const {container}=render(<Title/>);
    expect(container).toMatchSnapshot();
})

test('text checking',()=>{
    render(<Title/>)
    const text=screen.queryByTestId('title1').innerHTML;
    expect(text).toBe('32 Used Cars in India')
})
