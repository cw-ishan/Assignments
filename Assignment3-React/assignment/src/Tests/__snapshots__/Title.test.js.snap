// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing with mocking useSelector 1`] = `
<div>
  <h1
    data-testid="title1"
    style="font-size: 30px;"
  >
    32
     Used Cars in India
  </h1>
</div>
`;
